### PR TITLE
[Feature] ThreadPoolExecutor async fill_token_bitmask

### PR DESCRIPTION
**Pull request by ST-XX**
## Motivation

CPU-intensive operations like `fill_bitmask` can cause performance bottlenecks in the main thread. This PR converts the operation to an asynchronous model to improve overall system responsiveness and throughput by allowing other operations to continue while intensive computations are performed in the background.

## Modifications

- **Asynchronous Processing**: Converted `fill_bitmask` from a synchronous CPU-bound operation to an asynchronous task
- **Task Batching**: Implemented a configurable batching mechanism with a default  tasks per batch to optimize resource utilization
- **Join Mechanism**: Added join points at result consumption locations to ensure data is ready when needed
- **Configuration**: Added configuration options to control batch size and other async behavior parameters
- **Bug Fix**: Fixed JSON formatting issue that was producing consecutive spaces in output

## Usage or Command

```python
import openai

port = "8170"
client = openai.Client(base_url=f"http://127.0.0.1:{port}/v1", api_key="null")

completion = client.chat.completions.create(
    model="null",
    messages=[
        {
            "role": "user",
            "content": "Generate a JSON object containing: names of China's Four Great Inventions, their dynasties of origin, and brief descriptions (each under 50 characters)",
        }
    ],
    response_format={"type": "json_object"}
)
print(completion.choices[0].message.content)
```

## Accuracy Tests

The modifications maintain identical output results as the synchronous version. All existing test cases pass with the asynchronous implementation.

## Benchmark
dataset: xgrammar_bench
| Metric |  structured-output-ratio 1.0  | structured-output-ratio 0.0|
|---|---|---|
| Successful requests | 2000 | 2000 |
| Maximum request concurrency | 128 | 128 |
| Benchmark duration (s) | 81.95 | 87.75 |
| Total input tokens | 573,532 | 573,532 |
| Total generated tokens | 116,001 | 167,862 |
| Request throughput (req/s) | 24.41 | 22.79 |
| Output token throughput (tok/s) | 1,415.53 | 1,913.04 |
| Total Token throughput (tok/s) | 8,414.17 | 8,449.31 |
| Mean TTFT (ms) | 2,751.48 | 1,925.90 |
| Median TTFT (ms) | 2,672.84 | 1,856.67 |
| P99 TTFT (ms) | 6,078.50 | 4,580.82 |
| Mean TPOT (ms) | 40.60 | 42.35 |
| Median TPOT (ms) | 40.02 | 42.53 |
| P99 TPOT (ms) | 58.70 | 45.40 |
| Mean ITL (ms) | 38.66 | 41.73 |
| Median ITL (ms) | 38.39 | 41.35 |
| P99 ITL (ms) | 54.99 | 56.06 |
| Correct rate (%) | 94.30 | 76.15 |

## Checklist

- [x] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [x] Format your code, run `pre-commit` before commit.
- [x] Add unit tests. Please write the reason in this PR if no unit tests.
- [x] Provide accuracy results.
- [x] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/config.py [hunk 1]
```diff
@@ -1418,6 +1418,8 @@ def __init__(
         self.disable_any_whitespace: bool = True
         self.logits_processors: Optional[list[str]] = None
 
+        # async fill_bitmask, batch_size per worker
+        self.fill_bitmask_parallel_batch_size = 4
```
Comment (Jiang-Jia-Jun) [inline fastdeploy/config.py]: 增加一个envs.xxx来支持通过环境变量调整

**Review by Jiang-Jia-Jun — CHANGES_REQUESTED** [review]

File: fastdeploy/config.py (hunk 1 above)
Comment (ST-XX) [reply fastdeploy/config.py]: 已更新，添加环境变量

**Review by Jiang-Jia-Jun — APPROVED** [review]
